### ISSUE TITLE
navigator.mediaDevices suddenly undefined on Mac

### ISSUE THREAD

**Issue by boblund** [Winterreisender/webview-nodejs#22]
I have been using webview-nodejs @0.1.5 on a Mac (Ventura) for a WebRTC app. Suddenly, (today) it stopped working because navigator.mediaDevices is now undefined. The same app works in a browser. I did not touch the webview-nodejs node modules. Is it possible that webview-nodejs is now denied user media access? I looked in system settings/privacy and security/camera and I don't see a webview entry but I never had to add it here before. As an aside, I cannot add apps to access the camera.

I've shutdown and restarted, to no avail. Any suggestions on how to troubleshoot this are appreciated.

**Comment by boblund** [comment]
Turns out this is a Visual Studio Code problem accessing media devices. Running the app from the command line or starting vscode with sudo solves the problem. Why is just started happening is a mystery.

**Comment by Winterreisender** [comment]
Thanks for your feedback. I don't have any macOS devices so it's hard for me to test its compatiblility. 
Could you please tell me why you're using v0.1.5 instead of v0.4.0? Is v0.4.0 not working on macOS?

**Comment by boblund** [comment]
You're welcome. This port of webview for JavaScript is great; happy to help.

I tried installing 0.4.0 on a Mac M1 OSX 13.6.3 and got the error below (only first few lines)

```
npm ERR! code 1
npm ERR! path /.../node_modules/libwebview-nodejs
npm ERR! command failed
npm ERR! command sh -c cmake-js rebuild
```

0.1.5 didn't work out of the box but the changes to make it work are small

```
diff webview.js@0.1.5 webview.js@0.1.5.macM1
41c41,42
<     if (['linux', 'win32', 'darwin'].includes(platform) && arch == 'x64') {
---
>     if (['linux', 'win32', 'darwin'].includes(platform) && arch == 'x64'
> 	 || (platform == 'darwin' && arch == 'arm64')) {
```

and then

```
mv dist/libs/darwin/aarch64 dist/libs/darwin/arm64
```

**Comment by Winterreisender** [comment]
Thank! This should be a compiling issue and should be fix in the released v0.4.1. Please have a try if you're free.

**Comment by boblund** [comment]
Thanks. Just tried 0.4.1. It installed without a problem and I see the need for the M1 patch is gone. I'm doing a lot of testing of my app that uses this and will let you know if I find any issues.
